pkg/mount/mount.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2016 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mount
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/errwrap"
+)
+
+// GetMountsForPrefix returns all mountpoints associated with a process mount namespace,
+// filtered by prefix. If no prefixPath is specified, all mounts are returned.
+// The special value 0 as pid argument is used to specify the current process.
+func GetMountsForPrefix(prefixPath string, pid uint) (Mounts, error) {
```

Comment:
I'm usually not a fan of such approach, but in this case I'll try going this route as I already have some doubts that sorting on a prefix-filtered set produces the correct result, so `Mounts().Sorted().Filter(path)` should be a bit better.